File: Momento_2/Bloque_6/Animal/C++/src/Dog.cpp
#include "../include/Dog.h"

Dog::Dog() : Mammal() {}
Dog::Dog(std::string name) : Mammal(name) {}

std::string Dog::toString() const
{
    return "Dog [" + Mammal::toString() + " ]";
}

std::ostream &operator<<(std::ostream &os, const Dog &dog)
{
    os << dog.toString();
    return os;
}

File: Momento_2/Bloque_6/Animal/JAVA/src/Dog.java
public class Dog extends Mammal {
    public Dog() {
        super();
    }

    public Dog(String name) {
        super(name);
    }

    @Override
    public String toString() {
        return "Dog [" + super.toString() + " ]";
    }

    public void greets() {
        System.out.println("Woof!");
    }
}
